src/include/duckdb/main/config.hpp

```diff
@@ -189,6 +190,8 @@ struct DBConfig {
 	case_insensitive_map_t<std::unique_ptr<StorageExtension>> storage_extensions;
 	//! Extensions made to binder to implement the create_database functionality
 	vector<CreateDatabaseExtension> create_database_extensions;
+	//! Extensions made to binder to implement the drop_database functionality
+	vector<DropDatabaseExtension> drop_database_extensions;
```

Comment:
I wonder if we should pair up create and drop extensions into the same class. They are likely to always override both operators.

Also need to take another look at the virtual catalog PR we might want to plug in a little differently now that it landed